bbc/simorgh
# Topic Page fetches ATI Data from Simorgh BFF

Resolves JIRA [552](https://jira.dev.bbc.co.uk/browse/WSTEAMA-552)

Overall changes
======
- Returns Topic pages that fetch ATI data from the BFF within a metadata block:
``` "metadata": {
      "analytics": {
        "name": "pidgin.topics.c95y35941vrt.page",
        "producer": "PIDGIN"
      },
      "atiAnalytics": {
        "contentId": "urn:bbc:tipo:topic:c95y35941vrt",
        "contentType": "index-category",
        "pageIdentifier": "pidgin.topics.c95y35941vrt.page",
        "pageTitle": "Donald Trump"
      }
    }
   ```
- ATI data is fetched via the `genericPage` params builders
- `isMigrated` function now includes the `TOPIC` page for constructing the ATI URL 
- Various fixtures updated to reflect the new metadata block from the BFF
- Updates associated tests including async test for `getInitialData` which is needed to resolve the promise received when data from the BFF is fetched by Simorgh

Testing
======
- go to http://localhost:7080/pidgin/topics/c95y35941vrt
- see the ATI url appear in the network tab of the console with the required parameters

<img width="2020" alt="image" src="https://github.com/bbc/simorgh/assets/72317734/e89ac253-38d9-4b7b-aa9c-45695a02ee42">


Helpful Links
======

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)
